File: Arrays/sliding-window/max-sub-arr.go
package main


func maxSubArr(arr []int, k int) int {
	var maxSum int
	for i:=0; i <k; i++ {
		maxSum += arr[i]
	}
	var windowSum int = maxSum
	for i := k; i < len(arr); i++ {
		windowSum += arr[i] - arr[i-k]
		if windowSum > maxSum {
			maxSum = windowSum
		}
	}
	return maxSum
}


func main() {
 maxSubArr([]int{2,1,5,1,3,2}, 3)
}

File: Arrays/sliding-window/max-sub-arr.js
function maxSubArray(arr, k) {
  let maxSum = 0;

  for (let i = 0; i < k; i++) {
    maxSum += arr[i];
  }
  let windowSum = maxSum;
  for (let i = k; i < arr.length; i++) {
    windowSum = windowSum - arr[i - k] + arr[i];
    if (windowSum > maxSum) {
      maxSum = windowSum;
    }
  }
  return maxSum;
}

maxSubArray([1, 2, 3, 4, 5, 1, 2, 3], 3);
